### PR TITLE
Fixes and Improvments

### PR DESCRIPTION
**Pull request by pouyamn**
- Fixes '12:mm:ss am' display problem in strftime 
  - In 12 hour time, there is no hour 00, day begins with 12:00am after midnight.
- Adds support for '%-I' format token which were missed before
- Fixes '%X' token in strftime:
  - it should return 'HH:MM:SS' format but it returns 'HH:II:SS'!
- Fixes  Week of year, currently it starts with zero and increases every 7 days, ignoring weekdays
   - In Jalali Calendar, The first week of year is the week containing 1st of Farvardin and this week can be less than 7 days. So calculating week of year by dividing day-of-year by 7 was incorrect. For example, if 1st of Farvardin is Thursday, 3rd of Farvardin is Saturday and should be in week 2, but the code shows week 0. Well another point is that in Jalali calendar (and any other calendar that I have seen) week numbers start at 1 not 0.

- Refactor: `weekday` was rewritten to avoid having  lot's of `if` cases.

### CODE REVIEW

File: jdatetime/__init__.py [hunk 1]
```diff
@@ -492,28 +492,15 @@ def weekday(self):
...
-            return 2
-        if gd.weekday() == 1:
-            return 3
-        if gd.weekday() == 2:
-            return 4
-        if gd.weekday() == 3:
-            return 5
-        if gd.weekday() == 4:
-            return 6
+        return (gd.weekday() - 5) % 7
```
Comment (slashmili) [inline jdatetime/__init__.py]: Sweet!
Thanks for the improvement 👍 

Could you add test for this function that covers all week days?

File: jdatetime/__init__.py [hunk 2]
```diff
@@ -492,28 +492,15 @@ def weekday(self):
...
 
     def isoweekday(self):
         """Return the day of the week as an integer, where Shanbeh is 1 and Jomeh is 7"""
         return self.weekday() + 1
 
     def weeknumber(self):
         """Return week number """
-        return self.yday() // 7
+        return (self.yday() + date(self.year, 1, 1).weekday() - 1) // 7 + 1
```
Comment (slashmili) [inline jdatetime/__init__.py]: Has the old code have a bug? Could you add a test with the corner case that the old code doesn't support?

File: jdatetime/__init__.py [hunk 3]
```diff
@@ -577,10 +564,22 @@ def strftime(self, format):
         try:
             if self.hour > 12:
                 format = format.replace("%I", '%02.d' % (self.hour - 12))
-            else:
+            elif self.hour > 0:
```
Comment (slashmili) [inline jdatetime/__init__.py]: hmm it took me a while to understand this `if/else`.
What do you think if it's like

```python
if self.hour > 12:
    ...
elif self.hour == 12:
   ...
else:
   ...
```

Could you add a test that the old code was not covering?

File: jdatetime/__init__.py [hunk 4]
```diff
@@ -577,10 +564,22 @@ def strftime(self, format):
...
                 format = format.replace("%I", '%02.d' % (self.hour))
+            else:
+                format = format.replace("%I", '%02.d' % (12))
+        except:
+            format = format.replace("%I", '12')
+
+        try:
+            if self.hour > 12:
```
Comment (slashmili) [inline jdatetime/__init__.py]: this could be done in the above `if/else`

File: jdatetime/__init__.py (hunk 3 above)
Comment (pouyamn) [reply jdatetime/__init__.py]: Actually there is a better way, 
`self.hour % 12  or 12`
I'll use it in my next commit.

File: jdatetime/__init__.py (hunk 4 above)
Comment (pouyamn) [reply jdatetime/__init__.py]: I thought it would be more clear, the other tokens has been written in similar style.

File: jdatetime/__init__.py (hunk 2 above)
Comment (pouyamn) [reply jdatetime/__init__.py]: The old code was fundamentally wrong, it is always wrong there is no week 0 in any calendar system, it begins with 1.

> In Jalali Calendar, The first week of year is the week containing 1st of Farvardin and this week can be less than 7 days. So calculating week of year by dividing day-of-year by 7 was incorrect. For example, if 1st of Farvardin is Thursday, 3rd of Farvardin is Saturday and should be in week 2, but the code shows week 0. Well another point is that in Jalali calendar (and any other calendar that I have seen) week numbers start at 1 not 0.
And I have actually added it in `test.py` line 254.

File: jdatetime/__init__.py (hunk 1 above)
Comment (pouyamn) [reply jdatetime/__init__.py]: Done.

**Comment by slashmili** [comment]
Pushed as `3.5.0`

**Comment by slashmili** [comment]
Thanks @pouyamn 🎉 

Plz update your master if you are going to continue your awesome work on this library!
